Data/Graph/Inductive/Query/SP/DAG.hs

```diff
@@ -0,0 +1,200 @@
+-- Module written by Hunter Herman
+-- Contact @ hherman1@macalester.edu
+-- | An implementation of O(N + E) algorithms for finding shortest
+--   and longest paths on directed acyclic graphs. Most functions
+--   require the input of a topological sort of the graph and 
+--   the length of the topological sort. A topological sort can
+--   be obtained with the 'topsort' function in Data.Graph.Inductive.Query.DFS
+--
+--   It is important to note that the library does not check if
+--   the input graph is a DAG. Unreliable results will be returned on 
+--   any other graph. 
+--   
+--   It may be of interest to some that it is not necessary to output a list
+--   of Nodes from these functions. The shortest/longest path functions are simple
+--   combinations of runPaths, tracePath, and Data.Vector.maxIndex. 
+--
+--   The algorithms are adaptations of the following:
+--   <https://en.wikipedia.org/wiki/Topological_sorting#Application_to_shortest_path_finding>
+module Data.Graph.Inductive.Query.SP.DAG (
+      shortestPathBetweenNodes
+    , longestPathBetweenNodes
+    , longestPathFromSource
+    , longestPath
+    , tracePath
+    , runPaths
+    , Number
+) where
+
+import Prelude hiding (read)
```

Comment:
You should have the fgl-specific imports in a separate import block rather (either before or after the others; the current code-base has both but I'd prefer before) to help distinguish between the internal library imports and what's being used elsewhere.